crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -377,7 +405,21 @@ impl<'a> Submitter<'a> {
             let gas_limit = params.gas_estimate.to_f64_lossy() * ESTIMATE_GAS_LIMIT_FACTOR;
             let time_limit = target_confirm_time.saturating_duration_since(Instant::now());
             let gas_price = match estimator.estimate_with_limits(gas_limit, time_limit).await {
-                Ok(gas_price) => gas_price,
+                Ok(gas_price) => match pending_transaction {
+                    Some((_, pending_gas_price)) => {
+                        tracing::debug!("found pending transaction: {:?}", pending_transaction);
+                        let min_needed_gas_price = pending_gas_price.bump(1.125).ceil();
+                        pending_transaction = None;
+                        if gas_price.cap() >= pending_gas_price.cap()
+                            && gas_price.tip() >= pending_gas_price.tip()
```

Comment:
I might be misunderstanding (or missing some context) here but the surrounding code seems to suggest that `gas_price` should be compared against `min_needed_gas_price`, right?

Reply:
Oh, yes, nice catch. I should have covered this with unit tests...